src/Files.App/Utils/Storage/Operations/FilesystemHelpers.cs

```diff
@@ -161,11 +161,9 @@ public async Task<ReturnResult> DeleteItemsAsync(IEnumerable<IStorageItemWithPat
 
 			if (!permanently && registerHistory)
 				App.HistoryWrapper.AddHistory(history);
-
-			var itemsDeleted = history?.Source.Count ?? 0;
-
-			// Remove items from jump list
-			source.ForEach(async x => await jumpListService.RemoveFolderAsync(x.Path));
+			
+			// Await all removal tasks to complete concurrently
+			await Task.WhenAll(source.Select(x => jumpListService.RemoveFolderAsync(x.Path)));
```

Comment:
I'm not sure if we should await this. This is deleting items from the jump list, which is different from deleting the items itself, so it may be performed asynchronously.

Reply:
Agree